modules/core/src/norm.cpp

```diff
@@ -205,6 +205,773 @@ int normL1_(const uchar* a, const uchar* b, int n)
     return d;
 }
 
+int normDiffInf_(const uchar* a, const uchar* b, const uchar* mask, int* _result, int len, int cn)
+{
+    int i = 0;
+    int res = *_result;
+    int l = len*cn;
+
+    if(!mask)
+    {
+#if CV_SIMD
+        for(; i <= l - v_uint16::nlanes; i += v_uint16::nlanes )
+        {
+            v_int16 srca = v_reinterpret_as_s16(vx_load_expand(a + i));
+            v_int16 srcb = v_reinterpret_as_s16(vx_load_expand(b + i));
+
+            res = std::max(res, (int)v_reduce_max(v_abs(srca - srcb)));
+        }
+#endif
+        for(; i < l; i++)
+        {
+            res = std::max(res, (int)std::abs(a[i] - b[i]));
+        }
+    } else {
+#if CV_SIMD
+        int tres;
+#endif
+        if( cn == 1 )
+        {
+#if CV_SIMD
+            const v_int32 zero = vx_setall_s32((int)0);
+            for(; i <= len - v_uint16::nlanes; i+= v_uint16::nlanes)
+            {
+                v_int16 m = v_reinterpret_as_s16(vx_load_expand(mask + i));
+                v_int32 m0, m1;
+
+                v_expand(m, m0, m1);
+
+                m0 = m0 != zero;
+                m1 = m1 != zero;
+
+                v_int32 v_res = vx_setall_s32(res);
+
+                v_int16 srca = v_reinterpret_as_s16(vx_load_expand(a + i));
+                v_int32 srca0, srca1;
+
+                v_expand(srca, srca0, srca1);
+
+                v_int16 srcb = v_reinterpret_as_s16(vx_load_expand(b + i));
+                v_int32 srcb0, srcb1;
+
+                v_expand(srcb, srcb0, srcb1);
+
+                int max0 = v_reduce_max(v_select(m0, v_reinterpret_as_s32(v_abs(srca0 - srcb0)), v_res));
+                int max1 = v_reduce_max(v_select(m1, v_reinterpret_as_s32(v_abs(srca1 - srcb1)), v_res));
```

Comment:
Usually it is better to perform reduction outside (after) of the main SIMD loop.